lib/conduit_lwt_unix_ssl.ml

```diff
@@ -16,34 +16,23 @@
  *)
 
 open Lwt
+open Conduit_lwt_unix_common
 
 let _ = Ssl.init ()
 
-let safe_close t =
-  Lwt.catch
-    (fun () -> Lwt_io.close t)
-    (fun _ -> return_unit)
-
 let chans_of_fd sock =
-  let ic = Lwt_ssl.in_channel_of_descr sock in
-  let oc = Lwt_ssl.out_channel_of_descr sock in
+  let shutdown () = Lwt_ssl.ssl_shutdown sock in
+  let close () = Lwt_ssl.close sock in
+  let oc = Lwt_io.make ~mode:Lwt_io.output ~close:shutdown (Lwt_ssl.write_bytes sock) in
+  let ic = Lwt_io.make ~mode:Lwt_io.input ~close (Lwt_ssl.read_bytes sock) in
   ((Lwt_ssl.get_fd sock), ic, oc)
 
-let close (ic, oc) =
-  Lwt.join [ safe_close oc; safe_close ic ]
-
-let with_socket sockaddr f =
-  let fd = Lwt_unix.socket (Unix.domain_of_sockaddr sockaddr) Unix.SOCK_STREAM 0 in
-  Lwt.catch (fun () -> f fd) (fun e ->
-      Lwt.catch (fun () -> Lwt_unix.close fd) (fun _ -> return_unit) >>= fun () ->
-      fail e
-    )
-
 module Client = struct
   (* SSL TCP connection *)
-  let t = Ssl.create_context Ssl.TLSv1 Ssl.Client_context
+  let default_ctx = Ssl.create_context Ssl.SSLv23 Ssl.Client_context
+  let () = Ssl.disable_protocols default_ctx [Ssl.SSLv23]
```

Comment:
why this change from `Ssl.TLSv1` to `Ssl.SSLv23` followed by the disabling of the older protocol?